crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,387 @@
+    CssIdentifier, CssLanguage, CssSyntaxKind,
+};
+use biome_rowan::{AstNode, BatchMutationExt, NodeOrToken, SyntaxNode, TokenText};
+use biome_string_case::StrOnlyExtension;
+
+use crate::{
+    keywords::VENDOR_PREFIXES,
+    order::PROPERTY_ORDER_MAP,
+    utils::{get_longhand_sub_properties, get_reset_to_initial_properties, vender_prefix},
+    CssRuleAction,
+};
+
+declare_lint_rule! {
+    /// Enforce ordering of CSS properties.
+    ///
+    /// This rule checks if the properties and nested rules are in a consistent order.
+    ///
+    /// The expected ordering is roughly:
+    /// 1. Custom properties
+    /// 1. Layout properties (display, flex, grid)
+    /// 1. Margin & padding properties
+    /// 1. Typography properties (font, color)
+    /// 1. Interaction properties (pointer-events, visibility)
+    /// 1. Background & border properties
+    /// 1. Transition & animation properties
+    /// 1. Nested rules, media queries & other at-rules
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// p { color: black; display: block; }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// p { display: block; color: black; }
+    /// ```
+    ///
+    pub UseSortedProperties {
+        version: "next",
+        name: "useSortedProperties",
+        language: "css",
+        recommended: true,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+pub struct UseSortedPropertiesState {
+    block: CssDeclarationOrRuleBlock,
+    original_properties: Vec<AnyCssDeclarationOrRule>,
+    sorted_properties: RecessOrderProperties,
+    is_unsafe_to_sort: bool,
+}
+
+impl Rule for UseSortedProperties {
+    type Query = Ast<CssDeclarationOrRuleBlock>;
+    type State = UseSortedPropertiesState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let query_result = ctx.query();
+
+        let original_properties = query_result
+            .items()
+            .into_iter()
+            .collect::<Vec<AnyCssDeclarationOrRule>>();
+        let is_unsafe_to_sort = contains_shorthand_after_longhand(&original_properties)
+            || contains_unknown_property(&original_properties);
+        let sorted_properties = RecessOrderProperties::new(&original_properties);
+
+        Some(UseSortedPropertiesState {
+            block: query_result.clone(),
+            original_properties,
+            sorted_properties,
+            is_unsafe_to_sort,
+        })
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        if state.is_unsafe_to_sort || state.sorted_properties.is_sorted() {
+            return None;
+        }
+
+        return Some(RuleDiagnostic::new(
+            rule_category!(),
+            state.block.range(),
+            markup! {
+                "Properties can be sorted."
+            },
```

Comment:
Can you please add a second message? Rule diagnostics must follow our [rule pillars](https://biomejs.dev/linter/#rule-pillars). In this case, we need to explain why the sorting is good. We can probably say something like "Having a consistent order or CSS properties can improve the readability"